hack/setup-dev.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+root=$(dirname "${BASH_SOURCE[0]}")
+
+bash -e $root/install-dev-deps.sh
+source $root/dev-functions.sh
+
+# Create the control cluster
+kind delete cluster --name argocd-control-plane
+kind create cluster --name argocd-control-plane
+kubectl create namespace argocd
+
+# Setup argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj-labs/applicationset/v0.1.0/manifests/install.yaml
+
+# Login as admin and change def pass
+sleep 10
+kubectl wait --for=condition=ready --timeout=300s pod -l app.kubernetes.io/name=argocd-server -n argocd
+argoserver=$(kubectl get pods -n argocd -l app.kubernetes.io/name=argocd-server -o name | cut -d'/' -f 2)
+argocdlogin_initial="argocd login --insecure --username admin --password $argoserver argocd-server.argocd.svc.cluster.local:443"
+retry_argocd_exec "$argocdlogin_initial && argocd account update-password --account admin --current-password $argoserver --new-password admin"
+argocdlogin="argocd login --insecure --username admin --password admin argocd-server.argocd.svc.cluster.local:443"
+
+# # Create a new user
+kubectl apply -f $root/dev/users.yml 
+kubectl apply -f $root/dev/secrets.yml
+# and set its password
+retry_argocd_exec "$argocdlogin && argocd account update-password --account prc --current-password admin --new-password prc" || echo "Success"
+
+# Setup permissions for the new user
+kubectl apply -f $root/dev/perms.yml
+# Register in-cluster in argo secrets
+kubectl apply -f $root/dev/control-plane.yml
+
+# Create additional clusters to server as deployment targets
+register_argocd_cluster "prc-cluster-1" true
+register_argocd_cluster "prc-cluster-2" true
+
+# Generate token for the new user
+token=$(retry_argocd_exec "$argocdlogin >/dev/null && argocd account generate-token --account prc")
+serverip=$(kubectl get service -n argocd argocd-server -o=jsonpath='{.spec.clusterIP}')
+sleep 5
```

Comment:
same comment as above

Reply:
There are some race conditions of the exec process writing to stdout that might corrupt our echoing of token and server ip below. We can probably find the right level of redirecting stuff to /dev/null to avoid this, but I'd rather provide the ability to bootstrap your own env as soon as possible and iterate on resolving this sort of issues later.